### PR TITLE
Add doc string to dataset.py

### PR DESCRIPTION
**Pull request by smallv0221**
### PR types

Others 
### PR changes

Docs 
### Describe

Add doc string to dataset.py

### CODE REVIEW

File: paddlenlp/datasets/dataset.py [hunk 1]
```diff
@@ -86,12 +117,18 @@ def load_dataset(path_or_read_func,
 
 class MapDataset(Dataset):
     """
-    Wraps a dataset-like object as a instance of Dataset, and equips it with
-    `map` and other utility methods. All non-magic methods of the raw object
+    Wraps a map-style dataset-like object as a instance of Dataset, and equips it
```
Comment (yingyibiao) [inline paddlenlp/datasets/dataset.py]: a instance -> an instance

File: paddlenlp/datasets/dataset.py [hunk 2]
```diff
@@ -357,9 +418,11 @@ def read_datasets(self, splits=None, data_files=None):
     def read(self, filename, split='train'):
         """
         Returns an dataset containing all the examples that can be read from the file path.
-        If `self.lazy` is `False`, this eagerly reads all instances from `self._read()`
+
+        If `self.lazy` is False, this eagerly reads all instances from `self._read()`
         and returns an `MapDataset`.
```
Comment (yingyibiao) [inline paddlenlp/datasets/dataset.py]: returns an `MapDataset`. -> returns a `MapDataset`.

File: paddlenlp/datasets/dataset.py [hunk 3]
```diff
@@ -60,6 +61,36 @@ def load_dataset(path_or_read_func,
...
+
+    For all the names of datasets in PaddleNLP library, see here:  `dataset_list 
+    <https://paddlenlp.readthedocs.io/zh/latest/data_prepare/dataset_list.html>`__.
+
+    Args:
+        path_or_read_func (str|function): Name of of dataset in PaddleNLP library
+            or a custom data reading function.
+        name (str, optional): Additional name to select a more specific dataset.
+            Default to None.
+        data_files (str|list|tuple|dict, optional): Defineing to path of dataset
```
Comment (yingyibiao) [inline paddlenlp/datasets/dataset.py]: Defineing to-> Defining the

File: paddlenlp/datasets/dataset.py [hunk 4]
```diff
@@ -60,6 +61,36 @@ def load_dataset(path_or_read_func,
...
+    Args:
+        path_or_read_func (str|function): Name of of dataset in PaddleNLP library
+            or a custom data reading function.
+        name (str, optional): Additional name to select a more specific dataset.
+            Default to None.
+        data_files (str|list|tuple|dict, optional): Defineing to path of dataset 
+            files. Default to None.
+        splits (str|list|tuple, optional): Which split of the data to load.
+            Default to None.
+        lazy (bool, optional): Wheather to return `MapDataset` or a `IterDataset`.
```
Comment (yingyibiao) [inline paddlenlp/datasets/dataset.py]: or a -> or an

File: paddlenlp/datasets/dataset.py [hunk 5]
```diff
@@ -86,12 +117,18 @@ def load_dataset(path_or_read_func,
 
 class MapDataset(Dataset):
     """
-    Wraps a dataset-like object as a instance of Dataset, and equips it with
-    `map` and other utility methods. All non-magic methods of the raw object
+    Wraps a map-style dataset-like object as a instance of Dataset, and equips it 
+    with `map` and other utility methods. All non-magic methods of the raw object
     also accessible.
```
Comment (yingyibiao) [inline paddlenlp/datasets/dataset.py]: also accessible. -> are also accessible.

File: paddlenlp/datasets/dataset.py [hunk 6]
```diff
@@ -86,12 +117,18 @@ def load_dataset(path_or_read_func,
...
+
     Args:
-        data (list|Dataset): A dataset-like object. It can be a list or a
-            subclass of Dataset.
+        data (list|Dataset): An object with `__getitem__` and `__len__`. It could
```
Comment (yingyibiao) [inline paddlenlp/datasets/dataset.py]: with `__getitem__` and `__len__`. -> with `__getitem__` and `__len__` methods.

**Review by yingyibiao — COMMENTED** [review]
注意统一动词单复数

File: paddlenlp/datasets/dataset.py [hunk 7]
```diff
@@ -108,19 +145,27 @@ def _transform(self, data):
...
+        """
+        Basic function of `MapDataset` to get sample from dataset with a given 
+        index.
+        """
         return self._transform(self.new_data[
             idx]) if self._transform_pipline else self.new_data[idx]
 
     def __len__(self):
+        """
+        returns the number of samples in dataset.
```
Comment (ZeyuChen) [inline paddlenlp/datasets/dataset.py]: `returns` -> `Returns`, use capital letter.

**Review by ZeyuChen — COMMENTED** [review]
Few typo.

File: paddlenlp/datasets/dataset.py (hunk 7 above)
Comment (smallv0221) [reply paddlenlp/datasets/dataset.py]: Done.

File: paddlenlp/datasets/dataset.py [hunk 8]
```diff
@@ -86,12 +117,18 @@ def load_dataset(path_or_read_func,
 
 class MapDataset(Dataset):
     """
-    Wraps a dataset-like object as a instance of Dataset, and equips it with
-    `map` and other utility methods. All non-magic methods of the raw object
-    also accessible.
+    Wraps a map-style dataset-like object as an instance of Dataset, and equips it
```
Comment (yingyibiao) [inline paddlenlp/datasets/dataset.py]: Dataset -> `paddle.io.Dataset`

File: paddlenlp/datasets/dataset.py [hunk 9]
```diff
@@ -131,13 +176,14 @@ def filter(self, fn):
 
     def shard(self, num_shards=None, index=None):
         """
-        Use samples whose indices mod `index` equals 0 to update this dataset.
+        Uses samples whose indices mod `index` equals 0 to update this dataset.
+
         Args:
             num_shards (int, optional): A integer representing the number of
```
Comment (yingyibiao) [inline paddlenlp/datasets/dataset.py]: A integer -> An integer

File: paddlenlp/datasets/dataset.py [hunk 10]
```diff
@@ -131,13 +176,14 @@ def filter(self, fn):
...
                 data shards. If None, `num_shards` would be number of trainers.
                 Default: None
             index (int, optional): A integer representing the index of the
```
Comment (yingyibiao) [inline paddlenlp/datasets/dataset.py]: A integer -> An integer

File: paddlenlp/datasets/dataset.py [hunk 11]
```diff
@@ -187,12 +234,17 @@ def __getattr__(self, name):
 
 class IterDataset(IterableDataset):
     """
-    Wraps a dataset-like object as a instance of Dataset, and equips it with
+    Wraps a dataset-like object as an instance of Dataset, and equips it with
```
Comment (yingyibiao) [inline paddlenlp/datasets/dataset.py]: Dataset -> `paddle.io.Dataset`

File: paddlenlp/datasets/dataset.py [hunk 12]
```diff
@@ -255,13 +311,14 @@ def filter(self, fn):
 
     def shard(self, num_shards=None, index=None):
         """
-        Use samples whose indices mod `index` equals 0 to update this dataset.
+        Uses samples whose indices mod `index` equals 0 to update this dataset.
+
         Args:
             num_shards (int, optional): A integer representing the number of
```
Comment (yingyibiao) [inline paddlenlp/datasets/dataset.py]: A integer -> An integer

File: paddlenlp/datasets/dataset.py [hunk 13]
```diff
@@ -255,13 +311,14 @@ def filter(self, fn):
...
                 data shards. If None, `num_shards` would be number of trainers.
                 Default: None
             index (int, optional): A integer representing the index of the
```
Comment (yingyibiao) [inline paddlenlp/datasets/dataset.py]: A integer -> An integer

**Review by yingyibiao — APPROVED** [review]
LGTM

**Review by yingyibiao — APPROVED** [review]

**Pull request merged**